src/rules.ts

```diff
@@ -115,19 +115,39 @@ export const rules: Rule[] = [
       'Trailing spaces',
       'Removes extra spaces after every line.',
       RuleType.SPACING,
-      (text: string) => {
-        return text.replace(/[ \t]+$/gm, '');
+      (text: string, options = {}) => {
+        if (options['TwoSpaceLinebreak'] === false) {
+          return text.replace(/[ \t]+$/gm, '');
+        } else {
+          text = text.replace(/\b[ \t]$/gm, ''); // one whitespace
+          text = text.replace(/\b[ \t]{3,}$/gm, ''); // three or more whitespaces
+          text = text.replace(/\b( ?\t ?)$/gm, ''); // two whitespaces with at least one tab
```

Comment:
can you add tests for each of these cases in test.ts? Im not the best at verifying regex's.

Reply:
to be honest, I have no idea what exactly those tests do other than somehow checking that the before-example matches the after-example when the rule is applied. So I do not really know what I would have to do when you say "add tests for each of these cases" :(

(Sorry, as I once mentioned, I only know some basic coding I thought myself)

Reply:
No problem :)
The tests are exactly as you said, the check that applying a rule to the before text makes it into the after text. The format is very similar to an example, except they dont show up in the docs to avoid cluttering them. The purpose of the tests is to make sure edge cases are handled correctly.

So in this case, since there are no tests for this rule yet, add a describe statement to put tests into at https://github.com/platers/obsidian-linter/blob/master/src/test.ts#L256. 
Then add `it` statements to specify a behavior that the rule should have. Here you could add an it statement `it('Ignores 2 trailing spaces'...` with the before text containing lines with one whitespace, 3 or more whitespaces, and 2 whitespaces. This way we can verify the regex's work as intended. (basically its just a hidden example)

Let me know if you have any questions